### PR TITLE
chore(samples): Adds code examples to create, get, list, update and delete a SavedQuery

### PR DESCRIPTION
**Pull request by jeffreyaiatgoogle**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-asset/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: samples/snippets/quickstart_delete_saved_query_test.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google LLC.
```
Comment (dandhlee) [inline samples/snippets/quickstart_delete_saved_query_test.py:3]:
```suggestion
# Copyright 2022 Google LLC.
```

File: samples/snippets/quickstart_get_saved_query.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google LLC.
```
Comment (dandhlee) [inline samples/snippets/quickstart_get_saved_query.py:3]:
```suggestion
# Copyright 2022 Google LLC.
```

File: samples/snippets/quickstart_get_saved_query.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+def get_saved_query(saved_query_name):
+    # [START asset_quickstart_get_saved_query]
+    from google.cloud import asset_v1
+
+    # TODO saved_query_name = 'SavedQuery Name you want to get'
+
+    client = asset_v1.AssetServiceClient()
+    response = client.get_saved_query(request={"name": saved_query_name})
+    print("gotten_saved_query: {}".format(response))
```
Comment (dandhlee) [inline samples/snippets/quickstart_get_saved_query.py:29]:
```suggestion
    print(f"gotten_saved_query: {response}")
```

File: samples/snippets/quickstart_get_saved_query_test.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google LLC.
```
Comment (dandhlee) [inline samples/snippets/quickstart_get_saved_query_test.py:3]:
```suggestion
# Copyright 2022 Google LLC.
```

File: samples/snippets/quickstart_list_saved_queries.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google LLC.
```
Comment (dandhlee) [inline samples/snippets/quickstart_list_saved_queries.py:3]:
```suggestion
# Copyright 2022 Google LLC.
```

File: samples/snippets/quickstart_list_saved_queries.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+def list_saved_queries(parent_resource):
+    # [START asset_quickstart_list_saved_queries]
+    from google.cloud import asset_v1
+
+    # TODO parent_resource = 'Parent resource you want to list all saved_queries'
+
+    client = asset_v1.AssetServiceClient()
+    response = client.list_saved_queries(request={"parent": parent_resource})
+    print("saved_queries: {}".format(response.saved_queries))
```
Comment (dandhlee) [inline samples/snippets/quickstart_list_saved_queries.py:29]:
```suggestion
    print(f"saved_queries: {response.saved_queries}")
```

File: samples/snippets/quickstart_list_saved_queries_test.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google LLC.
```
Comment (dandhlee) [inline samples/snippets/quickstart_list_saved_queries_test.py:3]:
```suggestion
# Copyright 2022 Google LLC.
```

File: samples/snippets/quickstart_list_saved_queries_test.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+import os
+
+import quickstart_list_saved_queries
+
+PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
+
+
+def test_list_saved_queries(capsys):
+    parent_resource = "projects/{}".format(PROJECT)
```
Comment (dandhlee) [inline samples/snippets/quickstart_list_saved_queries_test.py:25]:
```suggestion
    parent_resource = f"projects/{PROJECT}"
```

File: samples/snippets/quickstart_create_saved_query_test.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
+SAVED_QUERY_ID = "saved-query-{}".format(uuid.uuid4().hex)
+
+
+def test_create_saved_query(capsys, saved_query_deleter):
+    saved_query = quickstart_create_saved_query.create_saved_query(
+        PROJECT, SAVED_QUERY_ID, "saved query foo")
+    saved_query_deleter.append(saved_query.name)
+    out, _ = capsys.readouterr()
+    assert "saved_query" in out
```
Comment (dandhlee) [inline samples/snippets/quickstart_create_saved_query_test.py:32]: All the tests in general only seem to be checking that RPC calls are made. The tests should check that (1) RPC calls are made, but also (2) check the contents of the response (delete doesn't return the response in the API so listing for empty queries or something would also work), to ensure that the tests are accurate.

File: samples/snippets/quickstart_update_saved_query.py [hunk 10]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google LLC.
```
Comment (dandhlee) [inline samples/snippets/quickstart_update_saved_query.py:3]:
```suggestion
# Copyright 2022 Google LLC.
```

File: samples/snippets/quickstart_update_saved_query.py [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    client = asset_v1.AssetServiceClient()
+    saved_query = asset_v1.SavedQuery()
+    saved_query.name = saved_query_name
+    saved_query.description = description
+    update_mask = field_mask_pb2.FieldMask()
+    # In this example, we only update description of the saved_query.
+    # You can update other content of the saved query.
+    update_mask.paths.append("description")
+    response = client.update_saved_query(request={"saved_query": saved_query, "update_mask": update_mask})
+    print("updated_saved_query: {}".format(response))
```
Comment (dandhlee) [inline samples/snippets/quickstart_update_saved_query.py:38]:
```suggestion
    print(f"updated_saved_query: {response}")
```

File: samples/snippets/quickstart_update_saved_query_test.py [hunk 12]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google LLC.
```
Comment (dandhlee) [inline samples/snippets/quickstart_update_saved_query_test.py:3]:
```suggestion
# Copyright 2022 Google LLC.
```

**Review by dandhlee — CHANGES_REQUESTED** [review]

File: samples/snippets/quickstart_get_saved_query.py [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    # [START asset_quickstart_get_saved_query]
```
Comment (dandhlee) [inline samples/snippets/quickstart_get_saved_query.py:30]: Snippet bot complaints:

```suggestion
    # [END asset_quickstart_get_saved_query]
```

File: samples/snippets/quickstart_create_saved_query.py [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+    # TODO project_id = 'Your Google Cloud Project ID'
+    # TODO saved_query_id = 'SavedQuery ID you want to create'
+    scope = "organizations/474566717491"  # Change to your org, folder. or project
+
+    client = asset_v1.AssetServiceClient()
+    parent = "projects/{}".format(project_id)
+    saved_query = asset_v1.SavedQuery()
+    saved_query.description = description
+    saved_query.content.iam_policy_analysis_query.scope = scope
+    # TODO: add more details into your saved query here.
```
[the remark this answers is not in the document]
Comment (jeffreyaiatgoogle) [reply samples/snippets/quickstart_create_saved_query.py:34]: I will add a link to the saved query content reference page to help user to compose their own saved query.

File: samples/snippets/quickstart_create_saved_query.py [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+import argparse
+
+
+def create_saved_query(project_id, saved_query_id, description):
+    # [START asset_quickstart_create_saved_query]
+    from google.cloud import asset_v1
+
+    # TODO project_id = 'Your Google Cloud Project ID'
+    # TODO saved_query_id = 'SavedQuery ID you want to create'
+    scope = "organizations/474566717491"  # Change to your org, folder. or project
```
[the remark this answers is not in the document]
Comment (jeffreyaiatgoogle) [reply samples/snippets/quickstart_create_saved_query.py:27]: I will remove the scope here and add a TODO comment later to let users to compose their saved query based on their need.

File: samples/snippets/quickstart_create_saved_query.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (dandhlee) [reply samples/snippets/quickstart_create_saved_query.py:34]: TODO's by itself can be left abandoned, could you file an issue as well? 😅

File: samples/snippets/quickstart_create_saved_query.py [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+    # TODO saved_query_id = 'SavedQuery ID you want to create'
+    client = asset_v1.AssetServiceClient()
+    parent = f"projects/{project_id}"
+    saved_query = asset_v1.SavedQuery()
+    saved_query.description = description
+
+    # TODO: customize your saved query based on the guide below.
+    # https://cloud.google.com/asset-inventory/docs/reference/rest/v1/savedQueries#IamPolicyAnalysisQuery
+    saved_query.content.iam_policy_analysis_query.scope = parent
+    saved_query.content.iam_policy_analysis_query.access_selector.permissions.append("iam.serviceAccounts.actAs")
```
Comment (dandhlee) [inline samples/snippets/quickstart_create_saved_query.py:35]: nit: line length
```suggestion
query_access_selector = saved_query.content.iam_policy_analysis_query.access_selector
query_access_selector.permissions.append("iam.serviceAccounts.actAs")
```

File: samples/snippets/quickstart_create_saved_query_test.py (hunk 9 above)
Comment (dandhlee) [reply samples/snippets/quickstart_create_saved_query_test.py]: Closing the loop: not in scope for this sample. Current checks are fine.

**Review by dandhlee — APPROVED** [review]
LGTM

**Comment by dandhlee** [comment]
Editing to `chore(samples)` to not trigger a release.

**Pull request merged**